### PR TITLE
feat: use the new Precompile in gnodev and in the addpkg/execution flow

### PR DESCRIPTION
**Pull request by moul**
*placeholder PR, will be updated soon*

* [x] `gnodev` tries to compile the translated .go code using go.
* [x]  bootstrap a simple `stdshim` package.
* [x] move `maths` to `gno.land/r/maths`
* [x] Precompiler tries to compile the translated .go code using go, before proceeding.
* [ ] way to ensure the go version we're checking against, for distributed determinism.

Note: I needed to fill the `examples/go.mod` to make `go` build do the work. I think that as soon as the PR will be merged, then we could rely entirely on standard go modules features.

Continues #119

### CODE REVIEW

**Comment by moul** [comment]
proof that it works -> it detects things like "unused variables".

Screenshot Manfred Macbook 2022-04-21 at 18 21 52@2x

**Comment by moul** [comment]
I've fixed all the issues I found except one about the `maths` package. I think that it shouldn't be placed in `stdlibs` but in p/maths instead, I'll make a dedicated commit for this and let me know if I can continue the PR or if I shouldn't do this. Thank you.

Screenshot Manfred Macbook 2022-04-21 at 19 13 36@2x

**Comment by moul** [comment]
with the math package moved, everything builds correctly with `gnodev`

Screenshot Manfred Macbook 2022-04-21 at 19 48 34@2x

File: examples/go.mod [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module github.com/gnolang/gno/examples
```
Comment (jaekwon) [inline examples/go.mod:1]: is this file needed?

Comment (moul) [reply examples/go.mod:1]: Right now yes, because of this -> https://github.com/gnolang/gno/pull/126/files/e41af05ad59b5ac45a8d192fdda32c52821a72bc#diff-cd249c062d77b4d6a5621756423850d0708ffbe4aae554ce33cdc5b908b0ee39R25

My goal is to make it unnecessary to have any go.mod, but I think that it will requires some adjustments and to merge all the precompile stuff to be able to use go mod’s cache effectively

FYI, I started a standalone repo to experience by myself Gno smart contracts from an external developer; my goal is to make it simpler over tie

File: nodes.go [hunk 2]
```diff
@@ -1080,6 +1080,10 @@ func ReadMemPackage(dir string, pkgPath string) *std.MemPackage {
 		if file.IsDir() {
 			continue
 		}
+		// skip files starting with a dot.
+		if strings.HasPrefix(file.Name(), ".") {
```
Comment (moul) [inline nodes.go:1084]: Not really related to this PR, but I think it’s appropriate

**Comment by moul** [comment]
@jaekwon I’ve refactored `gnodev` in a way I find pretty good today; and still a good base for improvements.

The simple precompile + syntax check is linked with the `gnokey maketx addpkg`; fyi, I know perform two different checks:
* a simple one file-per-file that checks basically that gofmt is working; this check is fast and does not depend on imports
* a more complete one that checks the full package (all .go in a folder) + run “go build”; this one is more complete, but in our case, it’s totally possible that the import change over time, or is not ready

I suggest to only enforce the simple `gofmt` check; and let the more complete one in `gnokey build`

File: examples/go.mod (hunk 1 above)
Comment (jaekwon) [reply examples/go.mod:1]: the link is broken
